script/wrapped-message-tests.js

```diff
@@ -0,0 +1,159 @@
+"use strict";
+
+var assert, config, errCount, Plugin, runNext, testCounter, tests;
+
+assert = require("assert");
+config = require("./manual-testing-config");
+errCount = 0;
+Plugin = require("../../../");
+testCounter = 0;
+tests = [
+    {
+        name: "heartbeat",
+        query: "INSERT INTO ?? (hash, isProcessing, owner, messageBase64, message, topic) VALUES('123', true, ?, '456', 'message', 'Test Topic');"
```

Comment:
This is a functional test, yes?  I would use the plugin and configure it to use a low heartbeat.  Then simulate what the clients would do.  Requeue is pretty easy - insert message, get message, requeue message, then listen again and get the same message.  Heartbeat failure may be a bit worse.  You could add a message, listen, then kill the heartbeat somehow (perhaps just run these things in a subprocess and kill the subprocess), then see if there are messages.  There shouldn't be.  Then wait and check again and the message should be available.